### PR TITLE
feat(xrpl): add Vault transactions, ledger entry, queries, and Lending Protocol updates

### PR DESCRIPTION
**Pull request by kpitapeersyst**
# feat(xrpl): add Vault transactions, ledger entry, queries, and Lending Protocol updates

## Description
This PR aims to add full support for the XRPL Vault feature (`VaultCreate`, `VaultSet`, `VaultDelete`, `VaultDeposit`, `VaultWithdraw`, `VaultClawback` transactions), the `Vault` ledger entry type, and the `vault_info` query, along with updates to Lending Protocol types (`Loan`/`LoanBroker` fields, `LoanPay` flags, `LoanBrokerCoverWithdraw` `DestinationTag`) and `int32` support in the binary codec.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactoring

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes

### binary-codec
  - Add int32 type support (for LoanScale field)
  - Add encode/decode tests for int32 LoanScale field
  - Suppress gosec G115 warning and add float64 edge case tests for int32

### xrpl/ledger-entry-types
  - Add Vault ledger entry type with support for XRP, IOU, and MPT assets
  - Extract Asset type to its own file and add MPT asset support with validation
  - Update Loan and LoanBroker ledger entry types with new fields

### xrpl/transaction
  - Add VaultCreate, VaultSet, VaultDelete, VaultDeposit, VaultWithdraw, and VaultClawback transactions with validation and tests
  - Add LoanPay flags, flag setter methods, and LoanBrokerCoverWithdraw DestinationTag support
  - Add MPT currency amount validation support
  - Refactor Flatten methods to use FlatTransaction return type
  - Fix Hash192 type usage for Vault ShareMPTID field
  - Add integration tests for vault transactions

### xrpl/queries
  - Add vault_info query with request validation for both RPC and WebSocket clients

### docs/examples
  - Add examples for vault operations and vault_info queries
  - Update changelog and documentation

## Notes (optional)

This PR introduces the core building blocks for the XRPL Lending Protocol vault feature. The Asset type was extracted from the AMM package into its own file to support reuse across Vault and other ledger entry types.

### CODE REVIEW

File: xrpl/transaction/vault_create.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	VaultCreateMaxDataLength = 512
+	// VaultCreateMaxMPTokenMetadataLength is the maximum length in characters for the MPTokenMetadata field (1024 bytes = 2048 hex chars).
+	VaultCreateMaxMPTokenMetadataLength = 2048
+	// VaultCreateMaxScale is the maximum value for Scale.
+	VaultCreateMaxScale = 18
+
+	// TfVaultPrivate indicates that the Vault is private.
+	TfVaultPrivate uint32 = 0x00010000
+	// TfVaultShareNonTransferable indicates that the Vault shares are non-transferable.
+	TfVaultShareNonTransferable uint32 = 0x00020000
```
Comment (JordiParraCrespo) [inline xrpl/transaction/vault_create.go]: We could export `VaultStrategyFirstComeFirstServe = 0x0001`

File: xrpl/ledger-entry-types/vault.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	// The asset of the vault. The vault supports XRP, IOU and MPT.
+	Asset Asset
+	// The total value of the vault.
+	AssetsTotal *types.XRPLNumber `json:",omitempty"`
+	// The asset amount that is available in the vault.
+	AssetsAvailable *types.XRPLNumber `json:",omitempty"`
+	// The potential loss amount that is not yet realized expressed as the vault's asset.
+	LossUnrealized *types.XRPLNumber `json:",omitempty"`
+	// The identifier of the share MPTokenIssuance object.
+	ShareMPTID types.Hash192
```
Comment (JordiParraCrespo) [inline xrpl/ledger-entry-types/vault.go]: 👁️

Comment (kpitapeersyst) [reply xrpl/ledger-entry-types/vault.go]: `ShareMPTID` is an `MPTokenIssuanceID`, which we are already using as `types.Hash192`
